corehq/warehouse/models/shared.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+from django.db import connections
+from django.conf import settings
+from django.template import Context, engines
+
+from corehq.sql_db.routers import db_for_read_write
+
+from corehq.warehouse.transforms import get_transform
+from corehq.warehouse.const import (
+    DJANGO_MAX_BATCH_SIZE,
+)
+
+
+class WarehouseTableMixin(object):
+
+    @classmethod
+    def sql_query_template_name(cls):
+        return None
+
+    @classmethod
+    def dependencies(cls):
+        raise NotImplementedError
+
+    @classmethod
+    def load(cls):
+        '''
+        Bulk loads records for a dim or fact table from
+        their corresponding dependencies
+        '''
+        for record in cls.record_iter():
+            print record
+
+    @classmethod
+    def _table_context(cls):
+        '''
+        Get a dict of slugs to table name mapping
+        :returns: Dict of slug to table_name
+        {
+            <slug>: <table_name>,
+            ...
+        }
+        '''
+        from corehq.warehouse.models import get_cls_by_slug
+
+        context = {cls.slug: cls._meta.db_table}
+        for dep in cls.dependencies():
+            dep_cls = get_cls_by_slug(dep)
+            context[dep] = dep_cls._meta.db_table
+        return context
+
+    @classmethod
+    def sql_query_template(cls, template_name):
+        path = os.path.join(
+            settings.BASE_DIR,
+            'corehq',
+            'warehouse',
+            'transforms',
+            'sql',
+            '{}.sql'.format(template_name),
```

Comment:
this dynamically assumes that there will be a sql file named the same thing as the slug to extract all the records it needs. i tend to prefer convention over configuration